OpenRA.Mods.Common/Activities/Repair.cs

```diff
@@ -81,6 +84,10 @@ public override Activity Tick(Actor self)
 				}
 
 				Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech", repairsUnits.Info.FinishRepairingNotification, self.Owner.Faction.InternalName);
+
+				foreach (var depot in host.Actor.TraitsImplementing<INotifyRepair>())
```

Comment:
There are a few more cases above where this can return `NextActivity` without running this notification.  If you override and put these in `OnFirstRun` and `OnLastRun` it should be a lot safer and cleaner.

Reply:
Indeed that is way more convenient. Updated.